_posts/2020-09-30-intro-to-hive-connector.md

```diff
@@ -0,0 +1,111 @@
+logic of the query engine that translates the SQL-esque Hive Query Language
+(HQL) into MapReduce jobs that run over files stored in the filesystem.
+_The storage_ component is simply that, it stores files in various formats and
+index structures to recall these files. The file formats can be anything as
+simple as JSON and CSV to more complex files such as columnar formats like ORC
+and Parquet. Traditionally, Hive ran on top of the Hadoop Distributed Filesystem
+(HDFS). As cloud-based options became more prevalent, object storage that used 
+Amazon's S3 interface become more and more common and Hive expanded to include
+storing files in Amazon S3, Azure Blob Storage, Google Cloud Storage (GCS), and
+others. In order for Hive to process these files, it must have a mapping from
+SQL tables in _the runtime_ to files and directories in _the storage_ component.
+To accomplish this, Hive uses _the metastore_ to keep the metadata about the
+files such as table columns, file locations, file formats, etc... 
+
+In the early days of Big Data systems, many expected query turnaround took a 
+long time because the data that ETL operated on was big. The primary goal in
+early iterations of these systems was simply throughput over large volumes of
+data while maintaining fault-tolerance. Over time, more businesses wanted to run
+fast interactive queries over their Big Data sets instead of running jobs that
+took hours and produced possibly undesirable results. Upon Presto's creation, 
+there were 300 petabytes of data and all the metadata for this data already
+existed. The data in storage was cumbersome to move, and the data in the
+metastore would have taken a long time to repopulate in another format. The
+metastore service still functioned properly and didn't need to be changed, and 
+only the runtime that executed Hive queries needed replacement. So the creators
+took this into account and built the Presto engine to utilize the existing
+metastore metadata and files residing in storage, and the presto runtime
+effectively replaced the Hive runtime responsible for analyzing the data.
+
+![](/assets/blog/intro-to-hive-connector/presto.png)
+
+Notice that the only change in the Presto architecture is the runtime, and
+the Hive metastore still exists along with the storage. This was not by
+accident. This design was to address a common pattern that many companies faced,
+and it simplified the migration from using Hive to using Presto. Since the 
+central model used to store the metadata makes use of the Hive metastore
+regardless of the storage component used, it seemed fitting that the connector 
+received the name Hive connector since the metastore can be configured over
+different storage solutions. 
+
+Where the confusion tends to come from, is when you search for a connector
+from the context of the storage systems you want to query and may not even be 
+aware the metastore is a necessity or even exists. You will typically look for
```

Comment:
Typically you look for an S3 connector, a GCS connector or a MinIO connector. All you need is the Hive connector and a Hive metastore to manage the metadata about the objects in your storage.